services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -205,17 +205,35 @@ public void run()
   {
     try {
       Injector injector = makeInjector();
-
+      final Server jettyServer = injector.getInstance(Server.class);
       try {
-        Lifecycle lifecycle = initLifecycle(injector);
-
+        final Lifecycle lifecycle = initLifecycle(injector);
+        Runtime.getRuntime().addShutdownHook(
+            new Thread(
+                new Runnable()
+                {
+                  @Override
+                  public void run()
+                  {
+                    log.info("Running shutdown hook");
+                    lifecycle.stop();
+                  }
+                }
+            )
+        );
         injector.getInstance(ExecutorLifecycle.class).join();
-        lifecycle.stop();
       }
       catch (Throwable t) {
         log.error(t, "Error when starting up.  Failing.");
         System.exit(1);
       }
+      log.info("Finished peon task");
+      try {
+        jettyServer.stop();
```

Comment:
is there a reason we need to shut down the jetty server separately? It should get stopped as part of the lifecycle.